src/java/org/apache/cassandra/cql3/Operator.java

```diff
@@ -359,20 +723,43 @@ public boolean isIN()
     }
 
     /**
-     * Checks if this operator is CONTAINS operator.
-     * @return {@code true} if this operator is a CONTAINS operator, {@code false} otherwise.
+     * Reverse this operator.
+     * @return the reverse operator from this operator.
+     */
+    public Operator negate()
+    {
+        throw new UnsupportedOperationException(this + " does not support negation");
+    }
+
+    /**
+     * Some operators are not supported by the read path because we never fully implemented support for them.
+     * It is the case for {@code IS_NOT} and {@code !=}
+     * @return {@code true} for the operators supported by the read path, {@code false} otherwise.
+     */
+    protected boolean isSupportedByReadPath()
+    {
+        return true;
+    }
+
+    /**
+     * The "LIKE_" operators are not real CQL operators and are simply a hack that should be removed at some point.
+     * Therefore, we want to ignore them for some operations.
+     * @return {@code true} for the "LIKE_" operators
      */
-    public boolean isContains()
+    private boolean isLikeXxx()
     {
-        return this == CONTAINS;
+        return name().startsWith("LIKE_");
```

Comment:
`name().startsWith(LIKE.name() + "_")` or sthg along those lines to avoid hardcoded values

Reply:
Using LIKE.name() is not really safer. The only thing safer is to have an actual check on each of the enums. I can do that if you want?

Reply:
I am not sure I follow with the 'safer' and 'checks' comment you make. I am only suggesting not using a hardcoded string in the code when we have the alternative of using the enum itself. It will make things much tidier, easy to follow and analyze in an IDE i.e.

Reply:
Why will it make thing easier to follow in the IDE?

Reply:
Bc you can easily find it's usages, navigation, no magic strings...

Reply:
As per conversation in Slack Benjamin is referring to the conceptual link between them. Whereas I am referring to the practicality of not having hardcoded magical values in the code. I can see both arguments, I'll favor authorship.